cas-server-support-x509/src/main/java/org/jasig/cas/adaptors/x509/authentication/handler/support/AbstractCRLRevocationChecker.java

```diff
@@ -58,19 +68,47 @@ public void check(final X509Certificate cert) throws GeneralSecurityException {
             throw new IllegalArgumentException("Certificate cannot be null.");
         }
         logger.debug("Evaluating certificate revocation status for {}", CertUtils.toString(cert));
-        final X509CRL crl = getCRL(cert);
-        if (crl == null) {
+        final Collection<X509CRL> crls = getCRLs(cert);
+
+        if (crls == null || crls.isEmpty()) {
             logger.warn("CRL data is not available for {}", CertUtils.toString(cert));
             this.unavailableCRLPolicy.apply(null);
             return;
         }
-        if (CertUtils.isExpired(crl)) {
-            logger.warn("CRL data expired on {}", crl.getNextUpdate());
-            this.expiredCRLPolicy.apply(crl);
+
+        final List<X509CRL> expiredCrls = new ArrayList<>();
+        final List<X509CRLEntry> revokedCrls = new ArrayList<>();
+
+        final Iterator<X509CRL> it = crls.iterator();
+        while (it.hasNext())  {
+            final X509CRL crl = it.next();
+            if (CertUtils.isExpired(crl)) {
+                logger.warn("CRL data expired on {}", crl.getNextUpdate());
+                expiredCrls.add(crl);
+            }
         }
-        final X509CRLEntry entry = crl.getRevokedCertificate(cert);
-        if (entry != null) {
-            throw new RevokedCertificateException(entry);
+
+        if (crls.size() == expiredCrls.size()) {
+            logger.warn("All CRLs retrieved have expired. Applying CRL expiration policy...");
+            for (final X509CRL crl : expiredCrls) {
+                this.expiredCRLPolicy.apply(crl);
+            }
+        } else {
+            crls.removeAll(expiredCrls);
+            logger.debug("Valid CRLs [{}] found that are not expired yet", crls);
+
+            for (final X509CRL crl : crls) {
+                final X509CRLEntry entry = crl.getRevokedCertificate(cert);
+                if (entry != null) {
+                    revokedCrls.add(entry);
+                }
+            }
+
+            if (revokedCrls.size() == crls.size() && !revokedCrls.isEmpty()) {
```

Comment:
I had a 2nd look at this. 

This check here:

`(revokedCrls.size() == crls.size() && !revokedCrls.isEmpty())`

This check here says: if all remaining crls (after checking and removing those that are expired) are considered revoked, then throw the exception and apply the policy. Otherwise, proceed. Because you may find that 5 crls are remaining, and only 2 of them have been revoked. So the 3 that remain are valid and we wouldn't need to apply the policy to those. The policy is only applied if everything is considered revoked; thus the check.

Does that make sense?